lib2/src/client.rs

```diff
@@ -0,0 +1,161 @@
+use std::convert::TryFrom;
+use std::sync::{Arc, Weak};
+
+#[cfg(feature="validator")]
+use validator::validator::Validator;
+use consensus::{
+    Consensus as AbstractConsensus,
+    AlbatrossConsensusProtocol,
+};
+use database::Environment;
+use network::NetworkConfig;
+use network_primitives::services::ServiceFlags;
+
+use crate::error::Error;
+use crate::config::{ClientConfig, ProtocolConfig};
+
+
+/// Alias for the Consensus specialized over Albatross
+pub type Consensus = AbstractConsensus<AlbatrossConsensusProtocol>;
+
+
+/// Holds references to the relevant structs. This is then Arc'd in `Client` and a nice API is
+/// exposed.
+///
+/// # ToDos
+///
+/// * Add `WalletStore` here.
+struct _Client {
+    /// The database environment. This is here to give the consumer access to the DB too. This
+    /// reference is also stored in the consensus though.
+    environment: Environment,
+
+    /// The consensus object, which maintains the blockchain, the network and other things to
+    /// reach consensus.
+    consensus: Arc<Consensus>,
+
+    /// The block production logic. This is optional and can also be fully disabled at compile-time
+    #[cfg(feature="validator")]
+    validator: Option<Arc<Validator>>
+}
+
+
+impl TryFrom<ClientConfig> for _Client {
+    type Error = Error;
+
+    fn try_from(config: ClientConfig) -> Result<Self, Self::Error> {
+        // create network config
+        // TODO: `NetworkConfig` could use some refactoring. So we might as well adapt it to the
+        // client API.
+
+        let mut network_config = match config.protocol {
+            ProtocolConfig::Dumb => {
+                NetworkConfig::new_dumb_network_config()
+            },
+            ProtocolConfig::Rtc => {
+                panic!("WebRTC is not yet implemented")
+            },
+            ProtocolConfig::Ws { host, port } => {
+                NetworkConfig::new_ws_network_config(host, port, false, config.reverse_proxy)
+            },
+            ProtocolConfig::Wss { host, port, pkcs12_key_file, pkcs12_passphrase } => {
+                let pkcs12_key_file = pkcs12_key_file.to_str()
+                    .unwrap_or_else(|| panic!("Failed to convert path to PKCS#12 key file to string: {}", pkcs12_key_file.display()))
+                    .to_string();
+                NetworkConfig::new_wss_network_config(host, port, false, pkcs12_key_file, pkcs12_passphrase)
+            }
+        };
+
+        // set user agent
+        network_config.set_user_agent(config.user_agent.into());
+
+        // initialize peer key
```

Comment:
We usually start comments with uppercase letter.
```suggestion
        // Initialize peer key.
```